### PR TITLE
Minor buffs for Heretic focuses / equipment and the Void Cloak.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- The Amber Focus is now fireproof.
- The Eldritch Medallion is now fireproof and acidproof, and also functions as a focus now. 
- The Void Cloak has been buffed in multiple ways. 
   - While unhooded, the Void Cloak now functions as a focus. 
   - The Void Cloak' s pockets can now hold more types items, including common ritual components.
   - The Void Cloak can now carry **one** sickly blade within its pockets. You can still also carry one on its suit storage. 
- Eldritch Potions are small now, instead of normal sized 
- Also cleaned up a bit of code involving the void cloak, and added an element for easily adding the focus behavior onto things. 
   - Some balloon alerts were added as well, just to be neat. 

## Why It's Good For The Game

"The Amber Focus is now fireproof."

The Amber Focus, despite being a magical artifact made of glass, very often _burned off_ of heretics and screwed them, which I found kind of lame. Acid is still available as people could play smart around using acid to disarm a heretic. 

"The Eldritch Medallion is now fireproof and acidproof, and also functions as a focus now."

Initially I wanted heretics to decide between opting for thermals or spellcasting, but I found that to be a lot more limiting than I originally thought. This made it a lot less desirable to go down the sidepath for the medallion. 
So, making it a focus (and not contend with your neck slot) makes it a bit more appealing. Acidproof was also added, as a sort of "upgrade" to the original focus. 
 
"The Void Cloak has been buffed in multiple ways."

The Void Cloak was in an interesting spot after the rework. As heretics have a lot less items to conceal, it's appeal changed from "conditionally, very interesting" to "almost never worth it". 
Expanding the potential of the pockets was a good start, but the key item a lot of heretics get caught by (their blade) was still sticking out for me. So, I thought it'd be neat if they could hide a single blade for emergencies in their suit pocket. 
Additionally, by conditionally making it a focus, heretics who decide to pick up one can easily swap between "stealth mode" and "combat mode" by flipping the hood up or down, which I think is an interesting dynamic. 

"Eldritch Potions are small now, instead of normal sized"

This one was more a sensibility thing. The Eldritch Flask is tiny, while the Eldritch Potions from the crucible are normal sized. Heretics who crossed into the Mawed Crucible often found themselves wanting to make  a ton of options, but not being able to carry more than two as they were so large. 

## Changelog

:cl: Melbert
balance: Heretic: The Amber Focus is now fireproof.
balance: Heretic: The Eldritch Medallion (thermal vision necklace) is now fireproof, acid proof, and works as a focus. 
balance: Heretic: The Void Cloak can carry more things in its pocket, including various ritual components (organs, bodyparts, flowers), small heretic items, and a singular sickly blade. It also functions as a focus while the hood is down. 
balance: Heretic: Mawed Crucible potions are now small sized (down from normal).
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
Glad to see you adopted the gamemode!

It needs as much love as it can get.

File: code/modules/antagonists/heretic/items/heretic_armor.dm [hunk 1]
```diff
@@ -31,16 +27,20 @@
 	. = ..()
 	if(!IS_HERETIC(user))
 		return
+	if(hood_up)
+		return
 
+	// Our hood gains the heretic_focus element.
 	. += span_notice("Allows you to cast heretic spells while the hood is up.")
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/heretic/items/heretic_armor.dm:34]: Why should this only show up when the hood is already up?

File: code/modules/antagonists/heretic/items/heretic_armor.dm [hunk 2]
```diff
@@ -50,36 +50,70 @@
...
+
+/obj/item/clothing/suit/hooded/cultrobes/void/examine(mob/user)
+	. = ..()
+	if(!IS_HERETIC(user))
+		return
+	if(!hood_up)
+		return
+
+	// Let examiners know this works as a focus only if the hood is down
+	. += span_notice("Allows you to cast heretic spells while the hood is down.")
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/heretic/items/heretic_armor.dm:78]: same here

File: code/modules/antagonists/heretic/items/heretic_armor.dm [hunk 3]
```diff
@@ -50,36 +50,70 @@
...
+	if(!IS_HERETIC_OR_MONSTER(wearer))
+		loc.balloon_alert(loc, "you can't get the hood up!")
 		return
 
-	to_chat(carbon_user,span_danger("You can't force the hood onto your head!"))
+	// When we make the hood, that means we're going invisible
+	make_invisible()
+	return ..()
+
+/// Makes our cloak "invisible". No the wearer, the cloak itself.
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/heretic/items/heretic_armor.dm:101]:
```suggestion
/// Makes our cloak "invisible". Not the wearer, the cloak itself.
```

File: code/modules/antagonists/heretic/items/heretic_armor.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply code/modules/antagonists/heretic/items/heretic_armor.dm:34]: that's fair

**Pull request merged**